flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/workflow/scheduler/EmbeddedQuartzScheduler.java

```diff
@@ -0,0 +1,228 @@
+            if (!quartzScheduler.checkExists(jobKey)) {
+                LOG.error("Failed to suspend a non-existent quartz schedule job: {}.", jobKey);
+                throw new SchedulerException(
+                        String.format(
+                                "Failed to suspend a non-existent quartz schedule job: %s.",
+                                jobKey));
+            }
+
+            quartzScheduler.pauseJob(jobKey);
+        } catch (org.quartz.SchedulerException e) {
+            LOG.error("Failed to suspend quartz schedule job: {}.", jobKey, e);
+            throw new SchedulerException(
+                    String.format("Failed to suspend quartz schedule job: %s.", jobKey), e);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public void resumeScheduleWorkflow(String workflowName, String workflowGroup)
+            throws SchedulerException {
+        JobKey jobKey = JobKey.jobKey(workflowName, workflowGroup);
+        lock.writeLock().lock();
+        try {
+            if (!quartzScheduler.checkExists(jobKey)) {
+                LOG.error("Failed to resume a non-existent quartz schedule job: {}.", jobKey);
+                throw new SchedulerException(
+                        String.format(
+                                "Failed to resume a non-existent quartz schedule job: %s.",
+                                jobKey));
+            }
+
+            quartzScheduler.resumeJob(jobKey);
+        } catch (org.quartz.SchedulerException e) {
+            LOG.error("Failed to resume quartz schedule job: {}.", jobKey, e);
+            throw new SchedulerException(
+                    String.format("Failed to resume quartz schedule job: %s.", jobKey), e);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public void deleteScheduleWorkflow(String workflowName, String workflowGroup)
+            throws SchedulerException {
+        JobKey jobKey = JobKey.jobKey(workflowName, workflowGroup);
+        lock.writeLock().lock();
+        try {
+            if (!quartzScheduler.checkExists(jobKey)) {
+                LOG.error("Failed to delete a non-existent quartz schedule job: {}.", jobKey);
+                throw new SchedulerException(
+                        String.format(
+                                "Failed to delete a non-existent quartz schedule job: %s.",
+                                jobKey));
+            }
+            quartzScheduler.deleteJob(jobKey);
+        } catch (org.quartz.SchedulerException e) {
+            LOG.error("Failed to delete quartz schedule job: {}.", jobKey, e);
+            throw new SchedulerException(
+                    String.format("Failed to delete quartz schedule job: %s.", jobKey), e);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    /** The {@link Job} implementation for embedded quartz scheduler. */
+    private class EmbeddedSchedulerJob implements Job {
```

Comment:
We will the SqlGatewayService in the next pr to support the refresh operation, but the SqlGatewayService can not be serialized to json string to Job, so  SqlGatewayService will be member of EmbeddedQuartzScheduler, we can refer it in Job directly, hence, it can not be changed to static class.